Tests/UIKitTests/UIImageExtensionsTests.swift

```diff
@@ -13,6 +13,40 @@ import XCTest
 import UIKit
 
 final class UIImageExtensionsTests: XCTestCase {
+    func testAverageColor() {
+        // note that not all colors precisely survive the roundtrip due to colorspace
+        // management. Use XCTAssertEqual w/ accuracy.
+        func assertColorsEqual(_ color1: UIColor, _ color2: UIColor, accuracy: CGFloat = 0.01) {
```

Comment:
This looks to me like a perfect extension to `XCTestCase` 😏

Reply:
You are 100% correct. 😃 Hesitant to jam that in this PR without a bit more direction, though. Where would it live? Is this for end users, or just our own tests? I don't want to slow down the PR further with build decisions...

Reply:
I ended spending a long time looking into this 😬 See #889

Reply:
Excellent, thank you! I'm glad I didn't try to attack this one because I'm sure I would've gotten stuck in build issues. Looks perfect, though. It's a bit hard to tell from the project diff - is this for internal or external use?

Let's get #889 merged and then I can revisit `testAverageColor`